libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationViewContext.kt

```diff
@@ -31,6 +48,63 @@ internal class DropInNavigationViewContext(
     var routeArrowOptions: RouteArrowOptions = RouteArrowOptions.Builder(context)
         .withAboveLayerId(RouteLayerConstants.TOP_LEVEL_ROUTE_LINE_LAYER_ID)
         .build()
+
+    val onBackPressedCallback = object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() = viewModel.onBackPressed()
+    }
+
+    val navigation: MapboxNavigation
+        get() = checkNotNull(MapboxNavigationApp.current()) {
+            "MapboxNavigationApp not initialized. You must call MapboxNavigationApp.setup() first."
+        }
+
+    //region Builders & Factories
+
+    fun routeOptionsBuilder() = RouteOptions.builder()
+        .applyDefaultNavigationOptions()
+        .applyLanguageAndVoiceUnitOptions(context)
+
+    fun mapAnnotationFactory() = MapMarkerFactory(
+        context,
+        BitmapMemoryCache(4 * MB_IN_BYTES)
+    )
+
+    //endregion
+
+    //region Use Cases
+
+    fun fetchRoutesUseCase() = FetchRouteUseCase(
```

Comment:
Please convert theses logic cases to something that is a `UIViewModel`

Reply:
"Can you migrate this logic to one of the `UIViewModel` subclasses"?

Yes I can.